### PR TITLE
fix(pr-monitor): push with explicit origin HEAD:<branch> refspec

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

- Bare `git push --force-with-lease` in `_rebase_source_branch` fails under `push.default=simple` when the local branch name differs from its upstream tracking ref — `PUSH_ERROR` logged, ready-flip skipped, PR left in draft for manual intervention.
- Fix: resolve the local branch via `git rev-parse --abbrev-ref HEAD`, push with explicit `origin HEAD:<branch>` refspec so the destination is unambiguous regardless of upstream-tracking config.
- Resolution failure logs a new structured `BRANCH_DETECT_ERROR` event and returns False — does **not** silently fall back to a hardcoded branch name.

Live evidence from PR #863's pr-monitor log:

```
[22:20:32] rebase: BASE branch=main
[22:20:32] rebase: PUSH_ERROR stderr=fatal: The upstream branch of your current branch does not match
                                        the name of your current branch.
                                        To push to the upstream branch on the remote, use
                                            git push origin HEAD:main
[22:20:32] ready: SKIPPED_REBASE_FAILED
```

## Test Plan

- [x] New `TestRebasePushRefspec` class — 3 regression tests:
  - Happy path asserts exact argv `["git", "push", "--force-with-lease", "origin", "HEAD:<branch>"]` and that `rev-parse` runs before push
  - `rev-parse` non-zero returncode → returns False, no push, `BRANCH_DETECT_ERROR` logged
  - `rev-parse` subprocess exception (`TimeoutExpired`) → returns False, no push, `BRANCH_DETECT_ERROR` logged
- [x] Existing `TestRebaseBeforeReady` tests updated to mock the new `rev-parse` call and assert the new push argv shape
- [x] All 58 tests in `tests/test_pr_monitor.py` pass
- [x] Full Python suite: 3 pre-existing failures on main (confirmed unrelated via `git stash` baseline)

## Verification

- [x] /gates passed
- [x] /verify completed (surface: workflow)
- [x] /review completed (findings: 3 suggestions, 0 critical/important — stale docstring addressed inline; detached-HEAD edge case and empty-stdout test gap are scope-creep for this surgical fix)

## Follow-up (not in this PR)

- Detached-HEAD edge case: if `rev-parse --abbrev-ref HEAD` returns the literal string `HEAD` (detached state), the guard doesn't catch it — unlikely to be reached after a successful rebase, but a defensive check would be cheap.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/pr_monitor.py [hunk 1]
```diff
@@ -532,9 +534,26 @@ def _run(cmd, timeout=60):
...
-        push = _run(["git", "push", "--force-with-lease"], timeout=60)
+        rev = _run(["git", "rev-parse", "--abbrev-ref", "HEAD"], timeout=10)
+    except (subprocess.TimeoutExpired, FileNotFoundError, OSError) as e:
+        logger.log("rebase", "BRANCH_DETECT_ERROR", reason=str(e))
+        return False
+    if rev.returncode != 0 or not rev.stdout.strip():
+        logger.log("rebase", "BRANCH_DETECT_ERROR",
+                   stderr=rev.stderr.strip()[:200])
+        return False
+    current = rev.stdout.strip()
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/pr_monitor.py]: Fixed in 346f308ee22adaf5e110c0db02bea2f08850f439 — Added detached HEAD guard: checks for literal 'HEAD' output from git rev-parse --abbrev-ref HEAD before using it as push refspec. Moved the strip() to a single assignment and distinguishes 'Detached HEAD' vs 'Empty output' in the error reason while preserving stderr on non-zero return.

[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/pr_monitor.py]: Not applicable — Already fixed in 346f308ee â€” scripts/pr_monitor.py:545-550 implements the exact suggestion: detached HEAD + empty output + non-zero returncode all return False with stderr logged, and rev.stdout.strip() is cached in `current`.
